src/argprocessing.cpp

```diff
@@ -242,6 +242,13 @@ process_arg(const Context& ctx,
     return nullopt;
   }
 
+  if (ctx.config.compiler_type() == CompilerType::cl
+      && util::starts_with(args[i], "/")) {
+    // MSVC understands both /option and -option, so convert all /option to
+    // -option to simplify our handling.
+    args[i][0] = '-';
```

Comment:
With this change, you break absolute paths without drive letters. Depending on what build system you are using, it is quite possible that a posix style path is used without a drive letter. 
Also, the feature should work when cross compiling from Linux to Windows with clang-cl.

Reply:
That doesn't seem to be the case:
C:\lo\ccache.git\build>cl -nologo -c /lo/ccache.git/build/a.cpp
cl : Command line warning D9002 : ignoring unknown option '/lo/ccache.git/build/a.cpp'
cl : Command line error D8003 : missing source filename

Reply:
But fair enough with clang-cl, I've added setting the slash back when the code gets to treating it as a file.

Reply:
Ok I didn't try it, I just looked it up in the Microsoft doc. (https://docs.microsoft.com/en-us/cpp/build/reference/cl-filename-syntax?view=msvc-160)
But I think it is very dangerous to just change the character here. Even if it does not work currently. 

What if the path is in quotes? Ccache should then receive the path without quotes.